pkg/api/job_analysis.go

```diff
@@ -2,82 +2,46 @@ package api
 
 import (
 	"fmt"
-	"net/http"
-	"strconv"
 	"time"
 
 	apitype "github.com/openshift/sippy/pkg/apis/api"
 	v1sippyprocessing "github.com/openshift/sippy/pkg/apis/sippyprocessing/v1"
 	"github.com/openshift/sippy/pkg/db"
+	"github.com/openshift/sippy/pkg/db/query"
 	"github.com/openshift/sippy/pkg/filter"
 )
 
 const PeriodDay = "day"
 const PeriodHour = "hour"
 
-type analysisResult struct {
+type AnalysisResult struct {
 	TotalRuns        int                                        `json:"total_runs"`
 	ResultCount      map[v1sippyprocessing.JobOverallResult]int `json:"result_count"`
 	TestFailureCount map[string]int                             `json:"test_count"`
 }
 
-type apiJobAnalysisResult struct {
-	ByPeriod map[string]analysisResult `json:"by_period"`
+type JobAnalysisResult struct {
+	ByPeriod map[string]AnalysisResult `json:"by_period"`
 }
 
-func PrintJobAnalysisJSONFromDB(w http.ResponseWriter, req *http.Request, dbc *db.DB, release string) {
-	fil, err := filter.ExtractFilters(req)
-	if err != nil {
-		RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{"code": http.StatusBadRequest, "message": "Could not marshal query:" + err.Error()})
-		return
+func PrintJobAnalysisJSONFromDB(
```

Comment:
Undertook a refactor here to get at the parts of this process I wanted, namely to get to the job IDs, but also just to generally clean up how far we pass the http request and response writer.